app/controllers/news_controller.rb

```diff
@@ -36,13 +45,46 @@ def show
 
 
   private
+  # An array of sections the user is filtering by, loaded by params
+  # with section ids
+  def set_filters
+    @filters = []
+    section_ids = Section.all.collect(&:id)
+
+    [:ministers, :departments].each do |filter_type|
+      if params[filter_type].present?
+        id_list = params[filter_type].map(&:to_i)
+
+        id_list.reject! do |item|
+          !section_ids.include? item
+        end
+
+        @filters += Section.where(id: id_list)
+      end
+    end
+  end
+
+
   def set_section
-    # silence these RecordNotFound exceptions in rollbar as they are generic 404s
-    begin
-      @section = SectionHome.find_by!(slug: params[:section]).section
-    rescue ActiveRecord::RecordNotFound => e
-      e.instance_variable_set(:@_rollbar_do_not_report, true)
-      raise
+    if params[:section].present?
+      begin
+        @section = SectionHome.find_by!(slug: params[:section]).section
+      rescue ActiveRecord::RecordNotFound => e
+        e.instance_variable_set(:@_rollbar_do_not_report, true)
+        raise
+      end
+    end
+  end
+
+
+  def filtered_articles
+    if @filters.present?
+      preload_news_articles
+        .includes(:sections)
```

Comment:
Pretty sure you don't need this line - it's already included. You can keep the `references` and extra `where` clause though.